vector.py

```diff
@@ -0,0 +1,113 @@
+from random import randint, random
+import math
+
+class Vector(object):
+
+    def __init__(self, x, y):
+        self.x = x
+        self.y = y
+
+    # str(a)
+    def __str__(self):
+        return "[" + str(self.x) + ", " + str(self.y) + "]"
+
+    # repr(a) or print(a)
+    def __repr__(self):
+        return self.__str__()
+
+    # add(b)
+    def add(self, b):
+        return Vector(self.x + b.x, self.y + b.y)
+
+    # a + b
+    def __add__(self, b):
+        return self.add(b)
+
+    # addAccumulate(b)
+    def addAccumulate(self, b):
+        self.x += b.x
+        self.y += b.y
+        return self
+
+    # a += b
+    def __iadd__(self, b):
+        return self.addAccumulate(b)
+
+    # sub(b)
+    def sub(self, b):
+        return Vector(self.x - b.x, self.y - b.y)
+
+    # a - b
+    def __sub__(self, b):
+        return self.sub(b)
+
+    # subAccumulate(b)
+    def subAccumulate(self, b):
+        self.x -= b.x
+        self.y -= b.y
+        return self
+
+    # a -= b
+    def __isub__(self, b):
+        return self.subAccumulate(b)
+
+    # mul(scalar)
+    def mul(self, scalar):
+        return Vector(self.x * scalar, self.y * scalar)
+
+    # a * scalar
+    def __mul__(self, scalar):
+        return self.mul(scalar)
+
+    # a *= scalar
+    def multAccumulate(self, scalar):
+        self.x *= x
+        self.y *= y
+        return self
+
+    # a *= scalar
+    def __imul__(self, scalar):
+        return self.multAccumulate(scalar)
+
+    # div(scalar)
+    def div(self, scalar):
+        return Vector(self.x / scalar, self.y / scalar)
+
+    # a / scalar
+    def __div__(self, scalar):
+        return self.div(scalar)
+
+    # divAccumulate(scalar)
+    def devAccumulate(self, scalar):
+        self.x /= scalar
+        self.y /= scalar
+        return self
+
+    # a /= scalar
+    def __idiv__(self, scalar):
+        return self.divAccumulate(scalar)
+
+    # get_mag()
+    def getMag(self):
+        hyp = math.sqrt((self.x ** 2) + (self.y ** 2))
+        return hyp
+
+    # norm()
+    def norm(self):
+        hyp = self.getMag()
+        if hyp != 0:
+            self.x /= hyp
+            self.y /= hyp
+            return Vector(self.x, self.y)
+
+    def setMag(self, value):
+        mag = self.getMag()
+        self.x *= value / mag
+        self.y *= value / mag
+        return Vector(self.x, self.y)
+
+    def limit(self, topSpeed):
```

Comment:
I'm saying it doesn't seem to fit the design. Limiting is a corner-case operation for the physics you are trying to emulate, not an inherent behavior of all mathematical vectors. Consider removing this method for now and if you really need it, creating a Physics utility library that includes it as a class method or something.